### PR TITLE
Add checkpoint restarts for ufs-cpld

### PR DESCRIPTION
**Pull request by DeniseWorthen**
## Description

- Adds ability to use checkpoint restarts for cpld restart tests.
- Drops c96mx025 restart test
- Adds restart test for cpld_control, cpld_control_c192 and cpld_bmark tests. These tests re-use the existing baselines.

### Issue(s) addressed

- Partially resolves Issue #316
- The existing setting of ``MOM6_REPRO_LA='True'`` in cpld_control_c384 and cpld_controlfrac_c384 prevent restart repro. This setting will need to be changed to 'False', which will change answers. Using the new default setting of ``vertex_shear=False`` in MOM6 did not resolve this requirement.
- Issue #343

## Testing

The new restart tests re-use existing baselines and should be b4b with existing baselines.

### CODE REVIEW

File: tests/fv3_conf/cpld_bmark_run.IN
Comment (junwang-noaa) [inline tests/fv3_conf/cpld_bmark_run.IN:66]: Do we still need this line as we copy the files in 72-92?

File: tests/fv3_conf/cpld_control_run.IN
Comment (junwang-noaa) [inline tests/fv3_conf/cpld_control_run.IN:58]: The same here. We might have extra ${RESTART_FILE_PREFIX}.* files in ./INPUT directory

Comment (DeniseWorthen) [reply tests/fv3_conf/cpld_control_run.IN:58]: Yes, I think that is right. I forgot to delete this line and also in the cpld_bmark_run.IN file.

Comment (DeniseWorthen) [reply tests/fv3_conf/cpld_control_run.IN:58]: Now that I look at it, I'm not quite sure. I tried to follow what is done in gfdlmp_run.IN. But I'm using a restart_interval of 6 hours so in a 1 day run, I get multiple timestamped FV3 restart files. I'm copying in here only the one which corresponds to the one I want to use in the restart test.

**Review by climbfuji — APPROVED** [review]
Regression tests passed as expected, changes look good from my side.

**Comment by DeniseWorthen** [comment]
All tests are completed and passed.

File: tests/fv3_conf/ccpp_csawmg_run.IN
Comment (junwang-noaa) [inline tests/fv3_conf/ccpp_csawmg_run.IN:18]: Thanks for fixing the issue

File: tests/fv3_conf/cpld_bmark_run.IN
Comment (junwang-noaa) [inline tests/fv3_conf/cpld_bmark_run.IN:62]: Maybe in the future we can consider changing "cice5_model" to "cice_model" to avoid confusion as we are running cice6?

Comment (DeniseWorthen) [reply tests/fv3_conf/cpld_bmark_run.IN:62]: Yes, I think we do need to do this. We can change it in the RT since the files are no longer sym-linked to the BM input directory that the workflow uses. But there is potential confusion if the workflow continues to use the name cice5 and the RT uses cice6.

**Review by junwang-noaa — APPROVED** [review]

**Comment by climbfuji** [comment]
Are we waiting for approval from @DusanJovic-NOAA (he is not listed as reviewer)? Or should we go ahead and merge (I can't).

**Comment by DeniseWorthen** [comment]
I've added him---the CI tests are complete now so we're ready. But I don't know how available Dusan is right now, so I'll give it a little bit and then merge if looks like he's out.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**
